vault/resource_database_secret_backend_connection.go

```diff
@@ -66,6 +66,34 @@ func databaseSecretBackendConnectionResource() *schema.Resource {
 				Sensitive:   true,
 			},
 
+			"elasticsearch": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Connection parameters for the elasticsearch-database-plugin.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"url": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The URL for Elasticsearch's API",
+						},
+						"username": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The username to be used in the connection URL",
+						},
+						"password": {
+							Type:        schema.TypeString,
+							Optional:    true,
```

Comment:
Should `url`, `username`, and `password` all be required? They're required by the engine itself noted [here](https://www.vaultproject.io/api/secret/databases/elasticdb.html#parameters).

Reply:
Sure, it just seemed like convention to have everything as optional. I though it might have something to do with multiple different types of optional blocks and so that their internals needed to be optional as well. That doesn't seem to be the case as I can run mysql acceptance tests with these fields as required. Pushing them.